src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -0,0 +1,33 @@
+from logging import getLogger
+from typing import Dict
+
+import pendulum
+
+logger = getLogger(__name__)
+
+
+class ProfileScaler:
+    """Class to scale existing profiles based on a new peak."""
+
+    def __init__(self, profile: Dict):
+        self._original_profile: Dict[pendulum.DateTime, float] = profile
+
+    def scale_by_peak(self, peak_kWh: float) -> Dict[pendulum.DateTime, float]:
+        """Return a profile obtained by scaling the original one using a new peak capacity."""
+        scaling_factor = self._compute_scaling_factor(peak_kWh)
+        return self._scale_by_factor(scaling_factor)
+
+    @property
+    def _original_peak_kWh(self) -> float:
+        return max(self._original_profile.values())
```

Comment:
This property will be evaluated every time that it is called. It will probably be much more performant to calculate the _original_peak_kWh in the constructor and maintain this as a class member variable, because that way it will be calculated only once.

Reply:
Done in [73fc176](https://github.com/gridsingularity/gsy-e/pull/1515/commits/73fc1761571e174bfcd4025216adb5ea7abfbdf0)